### PR TITLE
Add comments for calling out Edge module operations

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: iothub/device/src/IotHubModuleClient.cs [hunk 1]
```diff
@@ -190,16 +194,20 @@ public async Task SendEventBatchAsync(string outputName, IEnumerable<Message> me
...
-        /// A message handler can be unset by setting <paramref name="messageHandler"/> to null.
+        /// <para>
+        /// Calling this API more than once will result in the listener set last overwriting any previously set listener.
+        /// A listener can be unset by setting <paramref name="messageHandler"/> to null.
+        /// </para>
+        /// This API call is relevant for both IoT hub modules and IoT Edge modules.
         /// </remarks>
-        /// <param name="messageHandler">The delegate to be called when a message is sent to any input.</param>
-        /// <param name="userContext">generic parameter to be interpreted by the client code.</param>
+        /// <param name="messageHandler">The listener to be used when a could to device message is received by the client.</param>
```
Comment (timtay-microsoft) [inline iothub/device/src/IotHubModuleClient.cs:207]:
```suggestion
        /// <param name="messageHandler">The listener to be used when a cloud to device message is received by the client.</param>
```

**Review by patilsnr — APPROVED** [review]

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/device/src/IotHubModuleClient.cs (hunk 1 above)
Comment (brycewang-microsoft) [inline iothub/device/src/IotHubModuleClient.cs:207]:
```suggestion
        /// <param name="messageHandler">The listener to be used when a cloud-to-device message is received by the client.</param>
```

**Review by brycewang-microsoft — APPROVED** [review]

**Review by andyk-ms — APPROVED** [review]

File: iothub/device/src/IotHubBaseClient.cs [hunk 2]
```diff
@@ -131,10 +131,10 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         }
 
         /// <summary>
-        /// Sets a new delegate for the connection status changed callback. If a delegate is already associated,
-        /// it will be replaced with the new delegate.
+        /// Sets a new listener for the connection status change callback. If a listener is already associated,
+        /// it will be replaced with the new listener.
         /// </summary>
-        /// <param name="statusChangeHandler">The name of the method to associate with the delegate.</param>
+        /// <param name="statusChangeHandler">The listener for the connection status change callback.</param>
```
Comment (drwill-ms) [inline iothub/device/src/IotHubBaseClient.cs:137]: We should pick what term we like most and stick with it. Listener? Callback? Handler?

Across libraries, APIs, and comments/variable names.

Comment (abhipsaMisra) [reply iothub/device/src/IotHubBaseClient.cs:137]: That's a good idea. I like the term "listener". We had used it in our pnp-fied API comments, and it had been received well. So, I had been updating our APIs with that as I brought over the doc comments.

Comment (abhipsaMisra) [reply iothub/device/src/IotHubBaseClient.cs:137]: I'm not sure if it is very C#-ish though.

Comment (drwill-ms) [reply iothub/device/src/IotHubBaseClient.cs:137]: Inquire with the Azure SDK team?

Comment (abhipsaMisra) [reply iothub/device/src/IotHubBaseClient.cs:137]: This is the recommendation I received: https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#names-of-common-types
Based on this, callback would be the recommended suffix and term for our scenario.

Comment (abhipsaMisra) [reply iothub/device/src/IotHubBaseClient.cs:137]: https://github.com/Azure/azure-iot-sdk-csharp/pull/2760

**Pull request merged**